psl/psl-core/src/validate/validation_pipeline/validations/models.rs

```diff
@@ -240,7 +240,7 @@ pub(crate) fn primary_key_connector_specific(model: ModelWalker<'_>, ctx: &mut C
 }
 
 pub(super) fn connector_specific(model: ModelWalker<'_>, ctx: &mut Context<'_>) {
-    ctx.connector.validate_model(model, ctx.diagnostics)
+    ctx.connector.validate_model(model, ctx.relation_mode, ctx.diagnostics)
```

Comment:
Allowing access to `relation_mode` in the connectors implementing the `Connector` trait turned into a bit of diff noise, you can read the discussion happened before the implementation [here](https://prisma-company.slack.com/archives/C02EQD1EWAF/p1669210764461879).